inlang/source-code/versioned-interfaces/message/src/interface.ts

```diff
@@ -55,13 +77,44 @@ export const Variant = Type.Object({
 	pattern: Pattern,
 })
 
+export type InputDeclaration = Static<typeof InputDeclaration>
+export const InputDeclaration = Type.Object({
+	type: Type.Literal("input"),
+	name: Type.String(),
+
+	//TODO make this generic so that only Variable-Ref Expressions are allowed
+	value: Expression,
+})
+
+export type LocalDeclaration = Static<typeof InputDeclaration>
+export const LocalDeclaration = Type.Object({
+	type: Type.Literal("local"),
+	name: Type.String(),
+	value: Expression,
+})
+
+export type Declaration = Static<typeof Declaration>
+export const Declaration = Type.Union([LocalDeclaration, InputDeclaration])
+
+export type Translation = Static<typeof Translation>
```

Comment:
Translation vs. Locale - I think locales is the better term. It feels also more alinged with ICU [glossary](https://github.com/unicode-org/message-format-wg/blob/main/docs/glossary-and-resources.md#basic-terms-for-localization) compare source-locale and locale.
Since this property is used everywhere (also in fixtures for testing) this naming should be chosen wisely TBD.

Reply:
Loris will doublecheck the glossary and we will collect reasons for the term here